moved/src/state_actor.rs

```diff
@@ -450,6 +460,36 @@ impl<
         (outcome, receipts)
     }
 
+    fn simulate_transaction(&self, request: TransactionRequest) -> TransactionExecutionOutcome {
+        let mut tx = NormalizedEthTransaction::from(request.clone());
+        if let Some(from) = request.from {
+            tx.nonce = quick_get_nonce(&from.to_move_address(), self.state.resolver());
+        }
+        let tx = NormalizedExtendedTxEnvelope::Canonical(tx);
+
+        let block_header = HeaderForExecution {
+            number: self.height + 1,
+            timestamp: SystemTime::now()
+                .duration_since(UNIX_EPOCH)
+                .expect("Should get current time")
+                .as_secs(),
+            prev_randao: B256::random(),
+        };
+
+        match execute_transaction(
+            &tx,
+            &B256::random(),
+            self.state.resolver(),
+            &self.genesis_config,
+            0,
+            &(),
```

Comment:
We need the real impl for `BaseTokenAccounts` here otherwise the simulation will not correctly transfer tokens. Use `&self.base_token`.

Reply:
The issue with using `&self.base_token` is that if the user doesn't have any tokens it fails on `transfer_eth` even just to get a transaction estimate. Not sure if that's an expected behavior, so left it with `&()` to ignore any token transfer. What's the good compromise here?

Reply:
I think getting an error from `eth_estimateGas` when you have insufficient balance is expected behaviour. You could give it a try on an Ethereum testnet if you want.